### ISSUE TITLE
Some numeric literals do not parse

### ISSUE THREAD

**Issue by dxrcy** [rozukke/lace#29]
Integer literals are not recognized if they: are not prefixed by `#` or `(0)x`, or include both a negative sign and a hex prefix.

Forms which **do** parse:

```
x4      Positive hex
0x4     Positive hex full prefix
#4      Positive decimal with prefix
#-4     Negative decimal with prefix
```

Forms which **possibly should** parse (as they are accepted by other assemblers\*):

```
x-4     Negative hex
0x-4    Negative hex with full prefix
4       Positive decimal without prefix
-4      Negative decimal without prefix
-x4     Hex with negative sign before prefix
-0x4    Hex with negative sign before full prefix
-#4     Decimal with negative sign before prefix
```

*those being `laser` and `lc3as`

**Comment by rozukke** [comment]
Out of the forms suggested, I can accept `x-4` and `0x-4` since they can be useful in a limited set of cases, will work on getting them working. Allowing unsigned literals like `4` may potentially be allowed, but seems like an inconsistency since it is exclusively used by `.blkw` in most source examples. The rest feel a bit goofy and would also increase lexing complexity unnecessarily.

**Comment by dxrcy** [comment]
> The rest feel a bit goofy and would also increase lexing complexity unnecessarily.

I agree, they were just included for completeness.

**Comment by rozukke** [comment]
Closed with #34
